File: javascript/etl/etl.js
class ETL {
  transform(old) {
    const newData = {};
    for (const [score, letters] of Object.entries(old)) {
      letters.forEach((letter) => {
        newData[letter.toLowerCase()] = Number(score);
      });
    }
    return newData;
  }
}

module.exports = ETL;

File: csharp/etl/Etl.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class Etl
{
    public static IDictionary<string, int> Transform(IDictionary<int, IList<string>> old)
    {
        var newData = new Dictionary<string, int>();
        foreach (var entry in old)
        {
            foreach (var value in entry.Value)
            {
                newData[value.ToLower()] = entry.Key;
            }
        }
        return newData;
    }
}